### PR TITLE
Implement Mean(L(P) Norm Error)Loss

### PR DESCRIPTION
**Pull request by softmaximalist**
### Checklist

- [x] Confirmed that `cargo run-checks` command has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

[Image quality metrics #4312](https://github.com/tracel-ai/burn/issues/4312)

### Changes

Implemented a generalized Mean(L(P) Norm Error)Loss in the new file `crates/burn-nn/src/loss/mean_lp_error.rs`. It accepts positive p values where p = 1 results in MAE and p = 2 results in MSE. Additionally implemented 3 forward methods: one for no reduction, one where all elements are aggregated, and one where users can pass a slice of dimensions to reduce over.

### Testing

Added 18 test cases to `crates/burn-nn/src/loss/mean_lp_error.rs` covering different values for p, different tensor shapes, different forwards methods, and different reduction methods. 

### Note

The existing `crates/burn-nn/src/loss/mse.rs` file is not deleted as it can affect many users and existing code.

### CODE REVIEW

File: crates/burn-nn/src/loss/mean_lp_error.rs [hunk 1]
```diff
@@ -0,0 +1,622 @@
...
+/// let targets: Tensor<Backend, 2> = /* ground truth */;
+///
+/// // Compute loss with mean reduction
+/// let reduced_loss = mse.forward(predictions, targets, Reduction::Auto);
+///
+/// // Compute loss with no reduction
+/// let unreduced_loss = mse.forward_no_reduction(predictions, targets);
+/// ```
+#[derive(Module, Clone, Debug)]
+pub struct MeanLpErrorLoss {
```
Comment (laggui) [inline crates/burn-nn/src/loss/mean_lp_error.rs]: I think it would be more clear to go with a name like `LpLoss` instead.

1. The `Mean` prefix is not accurate, given that the loss can be parameterized by a `Reduction` arg
2. `Error` is a bit redundant with `Loss` when spelled out entirely

And thus adjust the doc accordingly.

**Review by laggui — CHANGES_REQUESTED** [review]
Thanks for the addition!

I suggest clearing up the name, and perhaps we could add some friendly constructors for standard losses e.g. `LpLossconfig::mae() / l1()` and `LpLossConfig::mse() / l2()` (not all four, but either mae/mse or l1/l2 combos).

Lmk what you think!

File: crates/burn-nn/src/loss/mean_lp_error.rs (hunk 1 above)
Comment (softmaximalist) [reply crates/burn-nn/src/loss/mean_lp_error.rs]: Great suggestions! I created two constructors l1() and l2() and renamed the structs to LpLoss and LpLossConfig. Also added new tests to make sure the new constructors work the same as setting p = 1.0 or p = 2.0.
